### PR TITLE
Fix the test_package

### PR DESCRIPTION
**Pull request by KyleFromKitware**
Run the server to go with the client so that the client has something to connect to and can run successfully.

### CODE REVIEW

File: test_package/greeter_server.cc [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    return Status::OK;
+  }
+
+ public:
+  Server* server;
+};
+
+void RunServer() {
+  std::string server_address("0.0.0.0:50051");
```
Comment (jtattermusch) [inline test_package/greeter_server.cc]: I think running a test that listens on a fixed port 50051 is pretty fragile.
If we really want to run the greeter example as a test that the conan package works fine (not a bad idea per se), we could do at least this:
- put greeter_server and greeter_client into just one binary.
- server would at least listen on a randomly selected port (bind to port "0") and the client would connect to that port. https://github.com/grpc/grpc/blob/b1547c9d19034740b1e5b4d73cb91d068dc0acb4/include/grpcpp/server_builder_impl.h#L133

File: test_package/CMakeLists.txt [hunk 2]
```diff
@@ -6,11 +6,7 @@ set(CMAKE_VERBOSE_MAKEFILE TRUE)
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup()
 
-if(NOT MSVC)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-else()
-  add_definitions(-D_WIN32_WINNT=0x600)
```
Comment (Croydon) [inline test_package/CMakeLists.txt:12]: Removing this breaks Windows builds, see the AppVeyor logs

File: test_package/CMakeLists.txt [hunk 3]
```diff
@@ -6,11 +6,7 @@ set(CMAKE_VERBOSE_MAKEFILE TRUE)
...
-endif()
+set(CMAKE_CXX_STANDARD 11)
```
Comment (Croydon) [inline test_package/CMakeLists.txt:12]: This is only available in CMake 3.1+
Please raise the cmake_minimum_required version to 3.1

**Comment by KyleFromKitware** [comment]
@Croydon are these test failures spurious?

**Comment by Croydon** [comment]
You can ignore CI errors which are caused by missing pre-builds, which is the case for GCC 9 and Apple-Clang 11 right now.

But Windows still needs fixing

**Comment by KyleFromKitware** [comment]
OK, let's see if this fixes the Windows issues.

**Comment by KyleFromKitware** [comment]
Two of the Windows builds failed due to a permission error when binding the server port. Any way to fix that?

**Review by jtattermusch — COMMENTED** [review]
> Two of the Windows builds failed due to a permission error when binding the server port. Any way to fix that?

I think the problem is relying of a fixed port 50051 (which might not be always available). Using a fixed port is very fragile. See my comment for a possible fix.

Possibly related: https://github.com/grpc/grpc/issues/19905

File: test_package/greeter_server.cc (hunk 1 above)
Comment (Croydon) [reply test_package/greeter_server.cc]: I'm also open to any other suggestion for the test_package. Whatever works best

**Comment by KyleFromKitware** [comment]
OK, I've merged the client and server into a single program with a randomly selected port.

**Review by Croydon — APPROVED** [review]
LGTM. 

Thank you!

**Pull request merged**